geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -287,22 +287,32 @@ public long zcount(SortedSetScoreRangeOptions rangeOptions) {
     return Coder.doubleToBytes(score);
   }
 
-  public static long zinterstore(RegionProvider regionProvider, RedisKey key,
-      List<ZKeyWeight> keyWeights,
-      ZAggregator aggregator) {
-    List<RedisSortedSet> sets = new ArrayList<>(keyWeights.size());
+  public static long zinterstore(RegionProvider regionProvider, RedisKey destinationKey,
+      List<ZKeyWeight> keyWeights, ZAggregator aggregator) {
+    boolean noIntersection = false;
 
+    List<RedisSortedSet> sets = new ArrayList<>(keyWeights.size());
     for (ZKeyWeight keyWeight : keyWeights) {
       RedisSortedSet set =
           regionProvider.getTypedRedisData(REDIS_SORTED_SET, keyWeight.getKey(), false);
 
+      // Need to loop through all keys to check for ERROR_WRONG_TYPE
+      if (noIntersection) {
+        continue;
+      }
+
       if (set == NULL_REDIS_SORTED_SET) {
-        return sortedSetOpStoreResult(regionProvider, key, new MemberMap(0), new ScoreSet());
+        noIntersection = true;
       } else {
         sets.add(set);
       }
     }
 
+    if (noIntersection) {
+      return sortedSetOpStoreResult(regionProvider, destinationKey, new MemberMap(0),
```

Comment:
Technically unrelated to this PR, but the arguments for the `sortedSetOpStoreResult()` method should probably be renamed. `MemberMap interMembers` and `ScoreSet interScores` should probably just be `MemberMap members` and `ScoreSet scores`. If it's not too much bother, would you mind renaming them, since you're already modifying this class?